File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.cpp
class MyQueue {
public:
    stack<int> in, op;

    MyQueue() {}

    void push(int x) { in.push(x); }

    int pop() {
        int rem = -1;
        if (!op.empty()) {
            rem = op.top();
            op.pop();
            return rem;
        } else {
            while (!in.empty()) {
                op.push(in.top());
                in.pop();
            }
        }
        rem=op.top();
        op.pop();
        return rem;
    }

    int peek() {

        if (!op.empty()) {
            return op.top();
        } else {
            while (!in.empty()) {
                op.push(in.top());
                in.pop();
            }
        }
        return op.top();
    }

    bool empty() { return in.empty() && op.empty(); }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.java
class MyQueue {

    Stack<Integer> input,output;
    int peekElement=-1;

    
    public MyQueue() {
        input=new Stack<>();
        output=new Stack<>();    
    }
    
    public void push(int x) {
        if(input.isEmpty()) peekElement=x;
        input.push(x);
    }
    
    public int pop() {
        if(output.isEmpty()){
            while(!input.isEmpty()){
                output.push(input.peek());
                input.pop();
            }
        }

        int x=output.peek();
        output.pop();
        return x;
    }
    
    public int peek() {
        if(!output.isEmpty()) return output.peek();
        return peekElement;
    }
    
    public boolean empty() {
        return input.isEmpty() && output.isEmpty();
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */